src/JuMP.jl

```diff
@@ -738,5 +729,19 @@ include("print.jl")
 # Nonlinear-specific code
 include("nlp.jl")
 
+
+typealias _getTypes Union{NonlinearExpression,NonlinearParameter, AffExpr}
+getValue{T<:_getTypes}(arr::Array{T}) = map(getValue, arr)
```

Comment:
The definitions like ``function $op{T<:JuMPTypes}(lhs::Number,rhs::Array{T})`` could easily be triggered by accident and lead to confusion, I think.